tests/scripts/csiaddons.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env -S bash +e
+
+# Copyright 2023 The Rook Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -xeEo pipefail
+
+CSIADDONS_VERSION="v0.8.0"
+CSIADDONS_CRD_NAME="csiaddonsnodes.csiaddons.openshift.io"
+CSIADDONS_CONTAINER_NAME="csi-addons"
+
+function setup_csiaddons() {
+    echo "setting up csi-addons"
+
+    echo "deploying controller"
+    kubectl create -f https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/$CSIADDONS_VERSION/deploy/controller/crds.yaml
+    kubectl create -f https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/$CSIADDONS_VERSION/deploy/controller/rbac.yaml
+    kubectl create -f https://raw.githubusercontent.com/csi-addons/kubernetes-csi-addons/$CSIADDONS_VERSION/deploy/controller/setup-controller.yaml
+
+    echo "enabling csi-addons"
+    kubectl patch cm rook-ceph-operator-config -n rook-ceph --type merge -p '{"data":{"CSI_ENABLE_CSIADDONS":"true"}}'
+
+    echo "setup csi-addons done!"
+}
+
+function verify_crd_created() {
+
+  crds=$(kubectl get crd "$CSIADDONS_CRD_NAME" -o=jsonpath="{.metadata.name}" 2>/dev/null)
+
+  if [ -n "$crds" ]; then
+    echo "CRD '$CSIADDONS_CRD_NAME' exists."
+  else
+    echo "CRD '$CSIADDONS_CRD_NAME' does not exist!"
```

Comment:
```suggestion
  if [ -n "$crds" ]; then
    echo "CRD '$CSIADDONS_CRD_NAME' exists."
  else
    echo "CRD '$CSIADDONS_CRD_NAME' does not exist!"
    exit 1
```
we are not doing exit in case of failure?